cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java

```diff
@@ -611,7 +608,19 @@ public void gitPullOrClone(Path target, String gitRepoUrl) {
         askToContinue(message);
       } else {
         pc.errorHandling(ProcessErrorHandling.WARNING);
-        result = pc.addArg("pull").run(false);
+        if (isOnline()) {
+          result = pc.addArg("fetch").addArg("origin").addArg("master").run(false);
```

Comment:
I am still not 100% sure if `git fetch ...` can reliable replace `git pull` nor am I really convinced that the source of our bug #6 is `git pull` itself but instead more something with Java invoking `git` and some stream redirection issue ending up in a blocked read operation from the process output.

Reply:
I'm pretty sure that git pull is the issue because if it needs to merge, it currently prompts for an input with a yes/no question and hangs the process. A solution for this could be to set a merge strategy beforehand. But I think the git fetch and especially reset should be more reliable here.

Reply:
I've adjusted the fetch to `master:master` now.

Reply:
I will test this once more, but also have to test the version with `force=false` what is tricky, as clone or updating of settings is not yet implemented/merged.